executor/grant.go

```diff
@@ -90,47 +92,83 @@ func (e *GrantExec) Next(ctx context.Context, req *chunk.Chunk) error {
 		}
 	}
 
-	// Grant for each user
-	for idx, user := range e.Users {
-		// Check if user exists.
+	// Commit the old transaction, like DDL.
+	if err := e.ctx.NewTxn(ctx); err != nil {
```

Comment:
It will commit the old transaction, but it also starts a new transaction. When will this new transaction be committed?

Reply:
There is a defer to set in transaction flag to false.
`defer func() { e.ctx.GetSessionVars().SetStatusFlag(mysql.ServerStatusInTrans, false) }()`